docs/reference/modules/node.asciidoc

```diff
@@ -32,42 +33,52 @@ configure this setting, the node has the following roles by default:
 * `remote_cluster_client`
 * `transform`
 
-NOTE: If you set `node.roles`, the node is assigned only the roles you specify.
+[IMPORTANT]
+====
+IMPORTANT: If you set `node.roles`, ensure you specify every node role your
+cluster needs. To access most {stack} features, a minimal cluster requires nodes
+with the following roles:
+
+- `master`
+- `ingest`
+- `transform`
+- `data`, `data_content`, or a combination of the `data_hot`, `data_warm`,
+  `data_cold`, and `data_frozen` roles
+====
```

Comment:
@jasontedor Are you comfortable with including this as a best practice?

The goal is to enable users to access most Stack features, such as Endpoint, if they set `node.roles`. The chief concern is that a user may not be aware that Endpoint or similar solutions require these roles otherwise.

Any other feedback you have is appreciated. Thanks!